api/v1alpha1/backend_types.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// KindBackend is the name of the Backend kind.
+	KindBackend = "Backend"
+)
+
+type AddressProtocolType string
+
+const (
+	// AddressProtocolTypeTCP defines the TCP address protocol.
+	AddressProtocolTypeTCP AddressProtocolType = "TCP"
+	// AddressProtocolTypeUDP defines the UDP address protocol.
+	AddressProtocolTypeUDP AddressProtocolType = "UDP"
+)
+
+type ApplicationProtocolType string
+
+const (
+	// ApplicationProtocolTypeH2C defines the HTTP/2 prior knowledge application protocol.
+	ApplicationProtocolTypeH2C ApplicationProtocolType = "H2C"
+	// ApplicationProtocolTypeWS defines the WebSocket over HTTP protocol.
+	ApplicationProtocolTypeWS ApplicationProtocolType = "WS"
```

Comment:
Should "WS" be a standalone ApplicationProtocolType?

Reply:
I tried to keep this in-sync with https://gateway-api.sigs.k8s.io/geps/gep-1911/#supporting-protocols